clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThread.java

```diff
@@ -254,8 +255,9 @@ private void closeInternal(final Duration timeout) {
     void cleanup() {
         log.trace("Closing the consumer network thread");
         Timer timer = time.timer(closeTimeout);
-        maybeAutoCommitAndLeaveGroup(timer);
+        maybeAutocommitOnClose(timer);
```

Comment:
After reviewing the ConsumerCoordinator, I split the original method into two functions: We should try to send an autocommit first and leave group last.